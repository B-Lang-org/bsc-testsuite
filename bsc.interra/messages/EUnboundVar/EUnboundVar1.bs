-----------------------------------------------------------------------
-- Project: Bluespec

-- File: EUnboundVar1.bs

-- Author : Nitin Chand Rahul      <ncr@noida.interrasystems.com>

-- Description: This testcase triggers an unbound variable error (EUnboundVar)

-- Error Message : bsc EUnboundVar1.bs
-- bsc: Compilation errors:
-- "EUnboundVar1.bs", line 31, column 12, Unbound variable: "clk"
-----------------------------------------------------------------------

package EUnboundVar1 (EUnboundVar1(..)) where
 
-- import Int

interface EUnboundVar1 = 
       dtoq   :: Int 1 -> Int 1 -> Bit 1 -> Action
       result :: Int 1

mkEUnboundVar1 :: Module EUnboundVar1
mkEUnboundVar1 =
        module

            d   :: Reg (Int 1)
            d <- mkRegU
            q   :: Reg (Int 1)
            q <- mkRegU
            clk ::  Bit 1

            rules
              "Assign":
                  when clk == 1
                   ==> action
                    q := d 
   
            interface
                dtoq ix iy iz = action
                                 d := ix
                                 q := iy
                               clk := iz
                        when clk == 1
                    
                result = q when clk == 1
