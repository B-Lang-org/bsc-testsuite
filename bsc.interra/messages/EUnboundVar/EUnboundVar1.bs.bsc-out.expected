checking package dependencies
compiling EUnboundVar1.bs
Error: "EUnboundVar1.bs", line 31, column 12: (T0004)
  Unbound variable `clk'
Error: "EUnboundVar1.bs", line 31, column 12: (T0020)
  Type error at:
    _tctemp1039

  Expected type:
    a__ b__

  Inferred type:
    Prelude.Bit 1

